src/me/goddragon/teaseai/api/picture/TagsFile.java

```diff
@@ -104,15 +104,17 @@ public boolean addTags(HashSet<PictureTag> tagsToAdd, File image) {
                     inputBuffer.append(strLine);
                     inputBuffer.append('\n');
                 }
-
+                TeaseLogger.getLogger().log(Level.INFO, "Added = " + added);
                 if (!added) {
                     String strLine = image.getName();
                     for (PictureTag tag : tagsToAdd) {
                         strLine += (" " + tag.getTagName());
                     }
-
-                    inputBuffer.append(strLine);
-                    inputBuffer.append('\n');
+                    if (!strLine.equals(image.getName())) {
```

Comment:
Why would you check that first? In the case of !added the file was not tagged previously which means we need to add it no matter whether the tags are empty right?
Because otherwise it will be an untagged and completely unknown image right?
You could also just check for "if(!added && !tagsToAdd.isEmpty())".

Reply:
This is because, if the strLine is just the name of the image, then it doesn't have a dress state or any tags and we don't want a picture in the tags file if it doesn't have any tags or a dress state.

Reply:
Why wouldn't we want an image without any tags inside the tags file? Does it matter?